### DISCUSSION TITLE
How can I add adaptive mesh to a model I created in Ls-PrePost

### DISCUSSION THREAD

**Discussion by ghost** [OpenRadioss/OpenRadioss#1977] [Q&A]
Hello OpenRadioss Community, my first post here!

For our projects to run efficiently and on time, we need to refine the mesh in deformation areas at least by 10 times. Meaning, I should start with elements 1 mm x 1 mm x 1 mm, but may end up in a very confined region by 0.1 x 0.1 x 0.1. 

I am using Ls-PrePost to prepare my models. Is there any chance I can only "run starter only" and manually add this rule? 

I am looking forward to hearing a solution!

Thanks and regards!

Erkin.

**Comment by PaulAltair** [comment]
I don't know a way to include this in a key/k input today, the Radioss admesh options aren't currently supported in *INCLUDE_RADIOSS, I'll see if we can add them

**Comment by ghost** [comment]
Looking forward to hearing a good news from you. I tried to run starter only, but it only gives one .rad file, which has only 10-12 lines. Adding there did not solve my problem.

**Reply by PaulAltair** [reply]
Yes, you won't get a starter by running starter only (only an engine) and the ADMESH (Radioss adaptive mesh cards) are starter keywords, we are looking into it

**Reply by sebastienVilleneuve** [reply]
Hi @Must-Erkin 

Debug is ongoing , at first look does not seem directly linked to ADMESH cards

Regards,
Sébastien

**Comment by sebastienVilleneuve** [comment]
Hi @Must-Erkin 

Can you test with 067609d6a564851f2e93c18593fff113abc47a4c ?

Regards,
Sébastien

**Comment by PaulAltair** [comment]
please find attached small example with ADMESH in a *RADIOSS_INCLUDE, now working, as of builds from this afternoon! (Thanks @sebastienVilleneuve )
[ADMESH_INCLUDE_RADIOSS.zip](https://github.com/OpenRadioss/OpenRadioss/files/12929163/ADMESH_INCLUDE_RADIOSS.zip)

**Reply by ghost** [reply]
I tried this one and it did not work. How do I include the re-meshing? Just put the K and include.inc in the same folder?

**Reply by ghost** [reply]
Hello Paul, I missed the portion where you said "builds from this afternoon". So after installing a newer build, it worked. Now my only new problem is that I cannot work this with multi-core solution:

ERROR ID :    704

** ERROR IN INPUT CHOICE

DESCRIPTION :  

   MESH ADAPTIVITY NOT AVAILABLE WHEN NSPMD > 1

**Comment by sebastienVilleneuve** [comment]
![test](https://github.com/OpenRadioss/OpenRadioss/assets/100404030/9711fe7f-95ad-4caf-be7e-21810fd2a440)

**Reply by ghost** [reply]
can you share the files of this one?

**Reply by ghost** [reply]
This is exactly what I would like to see, but I still cannot replicate this.

**Reply by ghost** [reply]
Please discard my earlier messages. I replicated this already. The only problem now is multi-core solution.

**Comment by ghost** [comment]
I tried this with solid elements and not working again, is there something else I should do in the include card?

**Comment by PaulAltair** [comment]
Sorry, I was on vacation a few days and didn't see your questions, /ADMESH functionality is only available with shell elements and as you found, isn't compatible with SPMD computation: see help documentation for further information [ADMESH in Help](https://help.altair.com/hwsolvers/rad/topics/solvers/rad/adaptive_meshings_overview_starter_r.htm)
What is the usecase for adaptive remeshing of Solids?

**Reply by ghost** [reply]
we are trying to simulate metal cutting, so we will need solid elements through thickness.

**Reply by PaulAltair** [reply]
Ok, If running from Radioss input format my suggestion for that type of model, would be to try to use sol2sph functionality for this. (Solids are broken into sph particle representation) Give me a day or 2 and I'll see if I can get a working example using the *INCLUDE_RADIOSS again so you can continue to use Dyna input for the majority of the model.

**Reply by PaulAltair** [reply]
I had a go at this, and it isn't possible right now I don't think from Dyna Format. I'll share a Radioss input format version in the next couple of days.

**Reply by PaulAltair** [reply]
I added a simple example to the model pages: [Sol2SPH example (Radioss Format)](https://openradioss.atlassian.net/wiki/spaces/OPENRADIOSS/pages/67796993/Sol2SPH+Ball+-+Plate+Impact)

**Reply by olosim** [reply]
Hi guys, are there any plans to include the /ADMESH functionality for solid elements in the future?

My application is metal forming such as the setting of a pop rivet including a transient spring back. I assume that if I know where high deformation will take place, it would be better to model these areas form the beginning with SPH?

**Reply by QiangF** [reply]
matafor can do remesh for severe deformation
http://metafor.ltas.ulg.ac.be/
